crates/xilem_masonry/src/sequence.rs

```diff
@@ -158,48 +201,91 @@ impl<State, Action, Marker, VT: ViewSequence<State, Action, Marker>>
-        for (i, (child, child_prev)) in self.iter().zip(prev).enumerate() {
-            // TODO: Do we want these ids to (also?) have a generational component?
-            let i: u64 = i.try_into().unwrap();
-            let id = NonZeroU64::new(i + 1).unwrap();
+        for (i, ((child, child_prev), (child_state, child_generation))) in self
+            .iter()
+            .zip(prev)
+            .zip(&mut seq_state.inner_with_generations)
+            .enumerate()
+        {
+            let id = create_vector_view_id(i, *child_generation);
             cx.with_id(ViewId::for_type::<VT>(id), |cx| {
-                let el_changed = child.rebuild(cx, child_prev, elements);
+                let el_changed = child.rebuild(child_state, cx, child_prev, elements);
                 changed.changed |= el_changed.changed;
             });
         }
         let n = self.len();
         if n < prev.len() {
             let n_delete = prev[n..].iter().map(ViewSequence::count).sum();
+            seq_state.inner_with_generations.drain(n..);
             elements.delete(n_delete);
             changed.changed |= ChangeFlags::CHANGED.changed;
         } else if n > prev.len() {
+            // Overflow condition: u32 incrementing by up to 1 per rebuild. Plausible if unlikely to overflow
+            seq_state.global_generation = match seq_state.global_generation.checked_add(1) {
+                Some(new_generation) => new_generation,
+                None => {
+                    // TODO: Inform the error
+                    tracing::error!(
+                        sequence_type = std::any::type_name::<VT>(),
+                        issue_url = "https://github.com/linebender/xilem/issues",
+                        "Got overflowing generation in ViewSequence. Please open an issue if you see this situation. There are known solutions"
+                    );
+                    // The known solution mentioned in the above message is to use a different ViewId for the index and the generation
+                    // We believe this to be superfluous for the default use case, as even with 1000 rebuilds a second, each adding
+                    // to the same array, this would take 50 days of the application running continuously.
+                    // See also https://github.com/bevyengine/bevy/pull/9907, where they warn in their equivalent case
+                    // Note that we have a slightly different strategy to Bevy, where we use a global generation
+                    // This theoretically allows some of the memory in `seq_state` to be reclaimed, at the cost of making overflow
+                    // more likely here. Note that we don't actually reclaim this memory at the moment.
+
+                    // We use 0 to wrap around. It would require extremely unfortunate timing to get an async event
+                    // with the correct generation exactly u32::MAX generations late, so wrapping is the best option
```

Comment:
Great writeup, and I agree, the probability that this happens per accident, is likely in the ballpark of a cryptographic hash collision... Not sure if this *could* be a security issue with a targeted attack, but I don't think we care at this point...